Docker/test/selenium/specs/upgrade/queryservice-pre-and-post-upgrade.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const assert = require( 'assert' );
+const { getElementByURI } = require( '../../helpers/blazegraph' );
+
+describe( 'Wikibase post upgrade', function () {
+
+	let oldItemID;
+	let oldPropertyID;
+
+	before( function () {
+		if ( process.env.RUN_QUERYSERVICE_POST_UPGRADE_TEST !== 'true' ) {
+			this.skip();
+		}
+	} );
+
+	it( 'Should be able find the item after upgrade', function () {
+
+		const result = browser.makeRequest(
+			process.env.MW_SERVER + '/w/api.php?action=wbsearchentities&search=UpgradeItem&format=json&language=en&type=item'
+		);
+		const success = result.data.success;
+		const searchResults = result.data.search;
+
+		assert( success === 1 );
+		assert( searchResults.length === 1 );
+		assert( searchResults[ 0 ].match.text === 'UpgradeItem' );
+		assert( searchResults[ 0 ].match.type === 'label' );
+
+		oldItemID = searchResults[ 0 ].id;
+
+		browser.url( process.env.MW_SERVER + '/wiki/Item:' + oldItemID );
+
+	} );
+
+	it( 'Should show up in Special:EntityData with json', function () {
+
+		const response = browser.makeRequest(
+			process.env.MW_SERVER + '/wiki/Special:EntityData/' + oldItemID + '.json'
+		);
+
+		const body = response.data;
+		const properties = Object.keys( body.entities[ oldItemID ].claims );
+
+		assert( properties.length === 1 );
+
+		oldPropertyID = properties[ 0 ];
```

Comment:
this is not ideal as it makes tests depend on the order they are run in. But it is something that can be addressed in a follow up PR.